sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/StatefulOperatorsHelperSuite.scala

```diff
@@ -0,0 +1,121 @@
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.streaming.state
+
+import org.apache.spark.sql.catalyst.expressions.{Attribute, SpecificInternalRow, UnsafeProjection, UnsafeRow}
+import org.apache.spark.sql.catalyst.expressions.codegen.GenerateUnsafeProjection
+import org.apache.spark.sql.execution.streaming.StatefulOperatorsHelper.StreamingAggregationStateManager
+import org.apache.spark.sql.streaming.StreamTest
+import org.apache.spark.sql.types.{IntegerType, StructField, StructType}
+
+class StatefulOperatorsHelperSuite extends StreamTest {
+  import TestMaterial._
+
+  test("StateManager v1 - get, put, iter") {
+    val stateManager = newStateManager(KEYS_ATTRIBUTES, OUTPUT_ATTRIBUTES, 1)
+
+    // in V1, input row is stored as value
+    testGetPutIterOnStateManager(stateManager, OUTPUT_ATTRIBUTES, TEST_ROW, TEST_KEY_ROW, TEST_ROW)
+  }
+
+  // ============================ StateManagerImplV2 ============================
+  test("StateManager v2 - get, put, iter") {
+    val stateManager = newStateManager(KEYS_ATTRIBUTES, OUTPUT_ATTRIBUTES, 2)
+
+    // in V2, row for values itself (excluding keys from input row) is stored as value
+    // so that stored value doesn't have key part, but state manager V2 will provide same output
+    // as V1 when getting row for key
+    testGetPutIterOnStateManager(stateManager, VALUES_ATTRIBUTES, TEST_ROW, TEST_KEY_ROW,
+      TEST_VALUE_ROW)
+  }
+
+  private def newStateManager(
+      keysAttributes: Seq[Attribute],
+      outputAttributes: Seq[Attribute],
+      version: Int): StreamingAggregationStateManager = {
+    StreamingAggregationStateManager.createStateManager(keysAttributes, outputAttributes, version)
+  }
+
+  private def testGetPutIterOnStateManager(
+      stateManager: StreamingAggregationStateManager,
+      expectedValueExpressions: Seq[Attribute],
+      inputRow: UnsafeRow,
+      expectedStateKey: UnsafeRow,
+      expectedStateValue: UnsafeRow): Unit = {
+
+    assert(stateManager.getValueExpressions === expectedValueExpressions)
+
+    val memoryStateStore = new MemoryStateStore()
+    stateManager.put(memoryStateStore, inputRow)
+
+    assert(memoryStateStore.iterator().size === 1)
+
+    val keyRow = stateManager.extractKey(inputRow)
+    assert(keyRow === expectedStateKey)
+
+    // iterate state store and verify whether expected format of key and value are stored
+    val pair = memoryStateStore.iterator().next()
+    assert(pair.key === keyRow)
+    assert(pair.value === expectedStateValue)
+    assert(stateManager.restoreOriginRow(pair) === inputRow)
+
+    // verify the stored value once again via get
+    assert(memoryStateStore.get(keyRow) === expectedStateValue)
+
+    // state manager should return row which is same as input row regardless of format version
+    assert(inputRow === stateManager.get(memoryStateStore, keyRow))
+  }
+
+}
+
+object TestMaterial {
+  val KEYS: Seq[String] = Seq("key1", "key2")
```

Comment:
Why are all of these in caps. Usually only simple constants like numbers or strings are in caps. This looks weird.

Also putting them in an object later make me jump back and forth too much to verify whats going on.